examples/image.md

```diff
@@ -0,0 +1,236 @@
+void SummaryWriter<Filewriter>::Image(const std::string& tag,
+                                      int step,
+                                      const std::vector<std::string>& encodedImages,
+                                      int height,
+                                      int width,
+                                      Filewriter& fw,
+                                      const std::string& displayName,
+                                      const std::string& description)
+```
+
+The API accepts `tag`, `step`, `encodedImage` (A vector of Image data in encoded format) and then the metadata about image along with filewriter object and Name and description.
+
+Following is a snippet that would log a multiple image for 1 step in temp directory.
+
+```cpp
+#include <mlboard/mlboard.hpp>
+#include <iostream>
+#include <chrono> 
+#include <ctime> 
+#include <future>
+
+int main()
+{
+    std::chrono::time_point<std::chrono::system_clock> start, end; 
+    start = std::chrono::system_clock::now(); 
+    mlboard::FileWriter f1("temp");
+    
+    ifstream fin("./assets/multiple_image.png", ios::binary);
+    ostringstream ss;
+    vector<string>encodedImages;
+    ss << fin.rdbuf();
+    encodedImages.push_back(ss.str());
+    fin.close();
+    fin.open("./assets/single_image.png", ios::binary);
+    ss << fin.rdbuf();
+    encodedImages.push_back(ss.str());
+    ss.str("");
+    fin.close();
+    mlboard::SummaryWriter<mlboard::FileWriter>::Image(
+         "Test Image", 1, encodedImages, 512, 512, f1, "Sample Image",
+         "This is a Sample image logged using mlboard.");
+    f1.Close();
+    end = std::chrono::system_clock::now(); 
+    std::chrono::duration<double> elapsed_seconds = end - start; 
+    std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
+    
+    std::cout << "finished computation at " << std::ctime(&end_time) 
+              << "elapsed time: " << elapsed_seconds.count() << "s\n"; 
+}
+```
+
+The output would be similar to: 
+
+<p>
+<img src = "assets/multiple_image.png" width = "800" height = "400"/>
+</p>
+
+### 3. Multiple Image Arma Mat
+
+Since we would be logging mlpack's metrics there are chances when we would have to visualize images stored in `arma::mat` and thus you can use the following API to directly log Armadillo matrices:
+
+```cpp
+void SummaryWriter<Filewriter>::Image(const std::string& tag,
+                int step,
+                arma::Mat<eT>& matrix,
+                mlpack::data::ImageInfo& info,
+                Filewriter& fw,
+                const std::string& displayName,
+                const std::string& description)
+```
+
+The API accepts `tag`, `step`, `encodedImage` (An `arma::mat` which has image data) and then the `mlpack::data::ImageInfo` object along with filewriter object and Name and description.
+
+Following is a snippet that would log a multiple image which is stored in `arma::mat` for 1 step in temp directory.
+
+```cpp
+#include <mlboard/mlboard.hpp>
```

Comment:
How did you compile the code? What was the command? I am getting a linking error with the below coomand
```
g++ multiple-image-arma.cpp -lproto -lprotobuf -lpthread -fopenmp -lmlpack -larmadillo -lboost_serialization -lboost_program_options -lboost_unit_test_framework -std=c++11

``` 

The error is
```
/tmp/ccEQMIxJ.o: In function `bool mlpack::data::Load<unsigned char>(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, arma::Mat<unsigned char>&, mlpack::data::ImageInfo&, bool)':
multiple-image-arma.cpp:(.text._ZN6mlpack4data4LoadIhEEbRKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EERN4arma3MatIT_EERNS0_9ImageInfoEb[_ZN6mlpack4data4LoadIhEEbRKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EERN4arma3MatIT_EERNS0_9ImageInfoEb]+0x17b): undefined reference to `mlpack::data::LoadImage(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, arma::Mat<unsigned char>&, mlpack::data::ImageInfo&, bool)'
multiple-image-arma.cpp:(.text._ZN6mlpack4data4LoadIhEEbRKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EERN4arma3MatIT_EERNS0_9ImageInfoEb[_ZN6mlpack4data4LoadIhEEbRKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EERN4arma3MatIT_EERNS0_9ImageInfoEb]+0x32d): undefined reference to `mlpack::data::LoadImage(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, arma::Mat<unsigned char>&, mlpack::data::ImageInfo&, bool)'
/tmp/ccEQMIxJ.o: In function `bool mlpack::data::Save<unsigned char>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, arma::Mat<unsigned char>&, mlpack::data::ImageInfo&, bool)':
multiple-image-arma.cpp:(.text._ZN6mlpack4data4SaveIhEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERN4arma3MatIT_EERNS0_9ImageInfoEb[_ZN6mlpack4data4SaveIhEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERN4arma3MatIT_EERNS0_9ImageInfoEb]+0x55): undefined reference to `mlpack::data::SaveImage(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, arma::Mat<unsigned char>&, mlpack::data::ImageInfo&, bool)'

```

Reply:
`g++ main.cpp -lproto -lprotobuf -lpthread -lmlpack -larmadillo --std=c++11`

Please make sure that while installing mlpack you enabled support for loading of image in mlpack

Reply:
`g++ main.cpp -lproto -lprotobuf -lpthread -lmlpack -larmadillo --std=c++11`

Please make sure that while installing mlpack you enabled support for loading of image in mlpack

Reply:
It seem so that it couldn't find function for loading of image

Reply:
> Please make sure that while installing mlpack you enabled support for loading of image in mlpack

How do we do that? Do we need to do it manually?

Reply:
fetch the master branch  of mlpack

create a build dir (let's start from fresh)

`cmake -DDOWNLOAD_STB_IMAGE=ON ../` (Please see if std is getting downloaded or not it would inform you that it is getting downloaded)

then usual `make` and `make install`

Reply:
https://github.com/mlpack/mlpack/blob/2567cc2e322a26ce7f262c479b0d3358c9b39ad3/CMakeLists.txt#L283

Reply:
Okay looks like the option to download stb is off now by default. Thanks

Reply:
https://github.com/mlpack/mlpack/blob/2567cc2e322a26ce7f262c479b0d3358c9b39ad3/CMakeLists.txt#L18 

Not sure if it was off by default, Currently it is On, May be you were trying out something else in a different branch and the installation was not the current one

Reply:
I am trying to install mlpack version of 3.3.2. latest one.
CMake Warning at CMakeLists.txt:362 (message):
  Could not download stb! Error code : "Unsupported protocol"! Error log:

Reply:
Okay I was able to resolve the issue. There was a problem with my CMake.